memtagabielf64/memtagabielf64.rst

```diff
@@ -0,0 +1,773 @@
+  capable hardware.
+
+.. note::
+
+   It is expected that stack instrumentation, because MTE instructions are not
+   part of the no-op space, may crash on non-MTE hardware or non-MemtagABI
+   linkers/loaders.
+
+General Principles
+------------------
+
+- In the general case, allocation tags are assigned at random at run-time. Some
+  tags may be "mostly random", e.g. for global variable or heap protection, it
+  may be useful to ensure that adjacent memory regions have different tags to
+  reliably detect linear buffer overflow. In addition, stack tagging
+  instrumentation may require some other non-fully-random optimisations, like
+  using a random base tag per-frame, but intra-frame objects have sequential
+  tags.
+
+General Restrictions
+--------------------
+
+- Global variables may only be tagged in dynamic executables, as assignment of
+  allocation tags happens during load time, and materialization of logical tags
+  happens during relocation processing.
+
+- Global variables may only be tagged on platforms that use RELA relocation
+  types. For more information, see the changed ``R_AARCH64_RELATIVE`` relocation
+  semantics description in this document.
+
+.. raw:: pdf
+
+   PageBreak
+
+Dynamic Section
+===============
+
+MemtagABI adds the following processor-specific dynamic array tags:
+
+.. table:: MemtagABI specific dynamic array tags
+
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | Name                            | Value      | d\_un  | Executable        | Shared Object     |
+  +=================================+============+========+===================+===================+
+  | ``DT_AARCH64_MEMTAG_MODE``      | 0x70000009 | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_HEAP``      | 0x7000000b | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_STACK``     | 0x7000000c | d\_ptr | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_GLOBALS``   | 0x7000000d | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_GLOBALSSZ`` | 0x7000000f | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+
+``DT_AARCH64_MEMTAG_MODE`` indicates the initial MTE mode that should be set. It
+has two possible values: ``0``, indicating that the desired MTE mode is
+Synchronous, and ``1``, indicating that the desired mode is Asynchonous. This
```

Comment:
What "Synchronous" and "Asynchronous" mean are not explained.

Reply:
Added some descriptions/references in the "Terms and Abbreviations" section.